### PR TITLE
Update build file with latest instructions

### LINKED ISSUE #144: Update BUILDING file
Our BUILDING file (https://github.com/esbmc/esbmc/blob/master/BUILDING) is outdated. 

New instructions are available at https://github.com/esbmc/esbmc/blob/master/.github/workflows/build.yml.

### PR DESCRIPTION
**Pull request by ericksonalves**
Our BUILDING file was outdated with current ESBMC build process. It also includes instructions on how to build the supported solvers.

This one closes #144

### CODE REVIEW

File: BUILDING.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
+# ESBMC Build Guide
+
+This is a guide on how to build ESBMC and its supported solvers.
+
+It has been tested with Ubuntu 19.10, but the steps are essentially the same for other Linux distributions.
```
Comment (lucasccordeiro) [inline BUILDING.md]: "essentially" -> "mostly"

File: BUILDING.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+Before starting, note that ESBMC is mainly distributed under the terms of the [Apache License 2.0](https://www.apache.org/licenses/LICENSE-2.0), so please make sure to carefully read it.
```
Comment (lucasccordeiro) [inline BUILDING.md]: "to carefully read it." -> "to read it carefully."

**Review by lucasccordeiro — APPROVED** [review]
Two tiny changes; overall, this looks good to me.

File: BUILDING.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+```
+tar xf clang+llvm-9.0.0-x86_64-linux-gnu-ubuntu-18.04.tar.xz && mv clang+llvm-9.0.0-x86_64-linux-gnu-ubuntu-18.04 clang9
+```
+
+## Setting Up Solvers
+
+ESBMC relies on SMT solvers to reason about formulae in its back-end.
+
+Currently we support the following solvers: __Boolector__, __CVC4__, __MathSAT__, __Yices__, __Yices (GMP)__, and __Z3__.
```
Comment (rafaelsamenezes) [inline BUILDING.md]: **YIces** and **Yices (GMP)** are the same, which is **Yices 2**

File: BUILDING.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+We have wrapped the entire build and setup of Z3 in the following command:
+
+```
+wget https://github.com/Z3Prover/z3/releases/download/z3-4.8.4/z3-4.8.4.d6df51951f4c-x64-ubuntu-16.04.zip && unzip z3-4.8.4.d6df51951f4c-x64-ubuntu-16.04.zip && mv z3-4.8.4.d6df51951f4c-x64-ubuntu-16.04 z3
+```
+
+If you need more details on Z3, please refer to [its Github](https://github.com/Z3Prover/z3).
+
+## Building ESBMC
```
Comment (rafaelsamenezes) [inline BUILDING.md:141]: I think this section could explain more about the cmake options for esbmc. Or at least point to the [options](https://github.com/esbmc/esbmc/blob/master/scripts/cmake/Options.cmake) module

**Review by rafaelsamenezes — CHANGES_REQUESTED** [review]

**Review by rafaelsamenezes — APPROVED** [review]
LGTM

**Review by lucasccordeiro — APPROVED** [review]

**Review by mikhailramalho — APPROVED** [review]

**Pull request merged**